packages/grafana-data/src/types/datasource.ts

```diff
@@ -251,13 +251,18 @@ abstract class DataSourceApi<
   getQueryDisplayText?(query: TQuery): string;
 
   /**
-   * Retrieve context for a given log row
+   * @deprecated please use DataSourceWithLogsContextSupport instead
    */
   getLogRowContext?: <TContextQueryOptions extends {}>(
     row: LogRowModel,
     options?: TContextQueryOptions
   ) => Promise<DataQueryResponse>;
 
+  /**
+   * @deprecated please use DataSourceWithLogsContextSupport instead
```

Comment:
Given some editors will complain about this (or better, they will show this method as deprecated while in fact is only getting moved) maybe a better description here explaining that the deprecation is only about the type definition in `DataSourceApi` and not about its functionality would be nice.

Also, maybe a note saying `if your editor shows a deprecation even after adding "implements DataSourceWithLogsContextSupport" you can ignore it`

Reply:
Changed in 81b401e8007769244e20e97c9eac634dfb417fc6